src/Files.App/Data/Factories/LocalizedEnumDescriptionFactory.cs

```diff
@@ -0,0 +1,92 @@
+		private static Dictionary<GridViewSizeKind, string> GridViewSizeKinds { get; } = [];
+		private static Dictionary<ColumnsViewSizeKind, string> ColumnsViewSizeKinds { get; } = [];
+
+		public static string Get(DetailsViewSizeKind value)
+		{
+			if (DetailsViewSizeKinds.Count == 0)
+			{
+				DetailsViewSizeKinds.Add(DetailsViewSizeKind.Compact, "Compact".GetLocalizedResource());
+				DetailsViewSizeKinds.Add(DetailsViewSizeKind.Small, "Small".GetLocalizedResource());
+				DetailsViewSizeKinds.Add(DetailsViewSizeKind.Medium, "Medium".GetLocalizedResource());
+				DetailsViewSizeKinds.Add(DetailsViewSizeKind.Large, "Large".GetLocalizedResource());
+				DetailsViewSizeKinds.Add(DetailsViewSizeKind.ExtraLarge, "ExtraLarge".GetLocalizedResource());
+			}
+
+			var stringValue = DetailsViewSizeKinds.GetValueOrDefault(value)!;
+			return stringValue;
+		}
+
+		public static string Get(ListViewSizeKind value)
+		{
+			if (ListViewSizeKinds.Count == 0)
+			{
+				ListViewSizeKinds.Add(ListViewSizeKind.Compact, "Compact".GetLocalizedResource());
+				ListViewSizeKinds.Add(ListViewSizeKind.Small, "Small".GetLocalizedResource());
+				ListViewSizeKinds.Add(ListViewSizeKind.Medium, "Medium".GetLocalizedResource());
+				ListViewSizeKinds.Add(ListViewSizeKind.Large, "Large".GetLocalizedResource());
+				ListViewSizeKinds.Add(ListViewSizeKind.ExtraLarge, "ExtraLarge".GetLocalizedResource());
+			}
+
+			var stringValue = ListViewSizeKinds.GetValueOrDefault(value)!;
+			return stringValue;
+		}
+
+		public static string Get(TilesViewSizeKind value)
+		{
+			if (TilesViewSizeKinds.Count == 0)
+			{
+				TilesViewSizeKinds.Add(TilesViewSizeKind.Small, "Small".GetLocalizedResource());
+			}
+
+			var stringValue = TilesViewSizeKinds.GetValueOrDefault(value)!;
+			return stringValue;
+		}
+
+		public static string Get(GridViewSizeKind value)
+		{
+			if (GridViewSizeKinds.Count == 0)
+			{
+				GridViewSizeKinds.Add(GridViewSizeKind.Small, "Small".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Medium, "Medium".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Three, "Three".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Four, "Four".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Five, "Five".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Six, "Six".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Seven, "Seven".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Large, "Large".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Nine, "Nine".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Ten, "Ten".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.Eleven, "Eleven".GetLocalizedResource());
+				GridViewSizeKinds.Add(GridViewSizeKind.ExtraLarge, "ExtraLarge".GetLocalizedResource());
```

Comment:
Can you match these up with the strings in the resource file?

Reply:
Yep. Did.